### PR TITLE
[Chore] Add AASM gem for guarding certain inquiry status updates

### PR DESCRIPTION
**Pull request by oliverochman**
PT https://www.pivotaltracker.com/story/show/178470013

### Changes proposed 
- Add events with AASM that changes the inquiry status
- Set restrictions on transitions between inquiry status
- Change param key for requests to Inquiries Controller

### Packages/Gems added
- AASM

### What I have learned working on this feature
- How to work with AASM events

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ gem 'slack-notify'
 gem 'devise_token_auth'
 gem 'pundit'
 gem 'pundit_helpers'
+gem 'aasm'
```
Comment (arunbhalli) [inline Gemfile:16]: aasm gem aams gem? because pr title is saying AAMS and you are using AASM.

Comment (oliverochman) [reply Gemfile:16]: you are correct, will change title and description

**Review by DavDan1 — APPROVED** [review]

**Review by 1CIM — APPROVED** [review]

**Review by LucasKnudsen — APPROVED** [review]

**Comment by LucasKnudsen** [comment]
Looks cool, interesting stuff with events

**Review by lukeperrin10 — APPROVED** [review]

**Review by arunbhalli — APPROVED** [review]

**Review by offcast-dev — APPROVED** [review]
State machines are great thing, awesome

**Pull request merged**